packages/react-client/src/ReactFlightClient.js

```diff
@@ -641,7 +641,15 @@ function createElement(
     }
     // TODO: We should be freezing the element but currently, we might write into
     // _debugInfo later. We could move it into _store which remains mutable.
-    Object.freeze(element.props);
+    if (
+      initializingChunkBlockedModel !== null &&
+      initializingChunkBlockedModel.deps > 0
```

Comment:
Since a cycle doesn't increase this dependency I'm not sure if a cycle that is in the props object itself would fail here. Maybe I'll just remove the deps check.